app/views/handlers/list_action.py

```diff
@@ -47,8 +49,33 @@ def handle_post(self):
             self.parent_location.block_id
         )
         self.questionnaire_store_updater.remove_answers(answer_ids_to_remove)
+        self.evaluate_sections()
         self.questionnaire_store_updater.save()
 
+    def evaluate_sections(self):
+        sections_to_evaluate = self._sections_to_evaluate()
+        section_paths = self._routing_paths(sections_to_evaluate)
+        for section_path in section_paths:
+            self._update_section_completeness(
+                Location(section_path[0][0]), section_path[1]
+            )
```

Comment:
I don't see the need for the _routing_paths method. This could just be:

```python
for section in self._sections_to_evaluate():
    path = self.path_finder.routing_path(section[0])
	self._update_section_completeness(
        Location(section_path[0][0]), section_path[1]
    )
```

Also, if the suggested changes are made to the `update_section_status` method in questionnaire store updater, this could become:

```python
for section_id, list_item_id in self._sections_to_evaluate():
    path = self.path_finder.routing_path(section_id)
	self.questionnaire_store_updater.update_section_status(
        is_complete = self.path_finder.is_path_complete(path),
        section_id=section_id,
        list_item_id=list_item_id
    )
```

Reply:
Initial suggestion is implemented - working on the changes to progress_store/questionnaire_store_updater